packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -692,8 +692,8 @@ class ReactDOMServerRenderer {
     // find the correct previous provider based on type
     let previousProvider;
     if (this.providerIndex > -1) {
```

Comment:
Seems like we can remove this condition? For loop already covers it.

Reply:
OK.